ratatui/examples/table.rs

```diff
@@ -202,13 +202,24 @@ impl App {
 
     fn draw(&mut self, frame: &mut Frame) {
         let vertical = &Layout::vertical([Constraint::Min(5), Constraint::Length(4)]);
-        let rects = vertical.split(frame.area());
+        let areas = vertical.split(frame.area());
 
         self.set_colors();
 
-        self.render_table(frame, rects[0]);
-        self.render_scrollbar(frame, rects[0]);
-        self.render_footer(frame, rects[1]);
+        let main_area = areas[0];
+        self.render_table(frame, main_area);
+        // Horizontal margin of 1 left and right
+        // Account for table header
+        let scrollbar_area = Rect {
+            x: main_area.x + 1,
+            y: main_area.y + 1,
+            width: main_area.width.saturating_sub(2),
+            height: (main_area.height.saturating_sub(1) / ITEM_HEIGHT as u16) * ITEM_HEIGHT as u16,
```

Comment:
I don't think I like this much, as it looks a bit weird when the track doesn't touch the border of the footer. But I also don't like the borders on the footer much aesthetically either.

The original intent of the margin was to make the scrollbar match the table highlight block a bit, but I think you're right in observing that the effect looks a bit janky right now.

Some Suggestions for de-janking this example a bit more if you're up for it...
- Make the track render as a solid bar
- Add a theme color for the track part
- Move the scrollbar to the extreme right side rather than being indented
- take up the whole table size with the scrollbar, not just the size of the items
- Remove the double borders on the footer and replace it with a solid background color fitting with the theme (take up 2 rows instead of 4)